recipes/libcudss/meta.yaml

```diff
@@ -0,0 +1,198 @@
+        - {{ pin_subpackage("libcudss-commlayer-nccl" ~ somajor, exact=True) }}
+        - {{ pin_subpackage("libcudss-commlayer-mpi" ~ somajor, exact=True) }}
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcudss.so.{{ soversion }}  # [linux]
+        - test -L $PREFIX/lib/libcudss.so.{{ somajor }}  # [linux]
+        - test ! -f $PREFIX/lib/libcudss_static.a  # [linux]
+        - if not exist %LIBRARY_BIN%\\cudss64_{{ somajor }}.dll exit 1  # [win]
+
+  - name: libcudss-dev
+    build:
+      run_exports:
+        - {{ pin_subpackage("libcudss" ~ somajor, max_pin="x.x.x") }}
+    files:
+      - lib/libcudss.so  # [linux]
+      - lib/cudss.lib  # [win]
+      - include/cudss*
+      - lib/cmake/cudss
+    requirements:
+      host:
+        - {{ pin_subpackage("libcudss" ~ somajor, exact=True) }}
+      run:
+        - {{ pin_subpackage("libcudss" ~ somajor, exact=True) }}
+      run_constrained:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    test:
+      files:
+        - test
+      requires:    # [build_platform == target_platform]
+        - {{ compiler("c") }}  # [build_platform == target_platform]
+        - {{ compiler("cxx") }}  # [build_platform == target_platform]
+        - {{ compiler('cuda') }}  # [build_platform == target_platform]
+        - {{ stdlib("c") }}  # [build_platform == target_platform]
+        - cmake   # [build_platform == target_platform]
+        - ninja  # [build_platform == target_platform]
+      commands:
+        - cmake ${CMAKE_ARGS} -GNinja test  # [build_platform == target_platform]
+        - cmake --build .  # [build_platform == target_platform]
+        - test -f $PREFIX/include/cudss.h  # [linux]
+        - test -f $PREFIX/include/cudss_distributed_interface.h  #[linux]
+        - test -L $PREFIX/lib/libcudss.so  # [linux]
+        - test ! -f $PREFIX/lib/libcudss_static.a  # [linux]
+        - if not exist %LIBRARY_LIB%\\cudss.lib exit 1  # [win]
+        - if not exist %LIBRARY_INC%\\cudss.h exit 1  # [win]
+        - if not exist %LIBRARY_INC%\\cudss_distributed_interface.h exit 1  # [win]
+
+# loadable modules; optional and only needed at runtime
+
+  - name: libcudss-commlayer-nccl
+    build:
+      skip: True  # [not linux]
+    requirements:
+      run:
+        - {{ pin_subpackage("libcudss-commlayer-nccl" ~ somajor, exact=True) }}
+    about:
+      summary: >-
+        The NCCL communications layer module for NVIDIA cuDSS
+
+  - name: libcudss-commlayer-nccl{{ somajor }}
+    build:
+      skip: True  # [not linux]
+    files:
+      - lib/libcudss_commlayer_nccl.so.*  # [linux]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - {{ compiler('cuda') }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - {{ stdlib("c") }}
+      host:
+        - cuda-version {{ cuda_compiler_version }}  # [cuda_compiler_version != "None"]
+        # FIXME: Get actual version ranges from kiril
```

Comment:
I need to know which minor version CUDSS is built against too. Otherwise, it is possible that CUDSS is compiled against a new NCCL before NCCL is released on conda-forge and then the NCCL binary will be missing a feature that CUDSS needs.

Reply:
It was NCCL 2.21(.5). I think we don't use any features above the major version 2.0.

Reply:
But i don't think we rely on any super-new features of NCCL (hopefully using an older version will not run into the fixed bugs since the last official release).